app/manager-frontend/src/components/forms/FormikSelect.jsx

```diff
@@ -25,13 +25,25 @@ export const FormikSelect = ({
   tooltip,
   alert,
   hasEmptyDefault,
+  sourceList,
```

Comment:
I'm still not convinced this is a desirable way to do this (hence ripping it out for the data sources and using plain strings).

Why do we need the form value to be an object? And more importantly why does this component need the entire list? or at least why does it need it differently to `options`?

I think the interface for this component should be `options`, which provide `value` and `label` for each item, and then it always writes `value` back to the Formik value.

Anything more complex should be dealt with by the parent component which uses it. if that results in a data lookup against a source list on that side when submitting the form, then so be it. but that's the responsibility of the user, not of this component. The user can pass the most useful value to lookup with later as the `value` in `options`; it's up to them.